modules/healthcheck/default.nix

```
@@ -0,0 +1,281 @@
+          };
+
+          statusReady = mkOption {
+            type = types.str;
+            default = "Service is ready.";
+            description = ''
+              The status message to send when the service is ready.
+              Use %OUTPUT% to substitute the output of the check command.
+            '';
+          };
+        };
+      };
+    in
+    {
+
+      config =
+        let
+          servicesWithHealthcheck = lib.filterAttrs (
+            _name: service: service.healthcheck != null
+          ) config.mcl.services;
+
+        in
+        {
+          assertions = lib.flatten (
+            lib.map (serviceName: {
+              assertion =
+                servicesWithHealthcheck."${serviceName}".healthcheck.readiness-probe.enable
+                && servicesWithHealthcheck."${serviceName}".healthcheck.exec != null;
+              message = "When healthcheck.readiness-probe is enabled, you must define `healthcheck.exec` with the service command.";
+            }) (lib.attrNames servicesWithHealthcheck)
+          );
+
+          systemd = {
+            timers = lib.mapAttrs' (
+              mainServiceName: serviceConfig:
+              let
+                cfg = serviceConfig.healthcheck;
+              in
+              {
+                name = "${mainServiceName}-liveness-check";
+                value = mkIf (cfg != null && cfg.liveness-probe.enable) (
+                  let
+                    probe = cfg.liveness-probe;
+                  in
+                  {
+                    description = "Timer for ${mainServiceName} liveness probe";
+                    timerConfig = {
+                      Unit = "${mainServiceName}-liveness-check.service";
+                    };
+                    wantedBy = [ "${mainServiceName}.service" ];
+                  }
+                );
+              }
+            ) (servicesWithHealthcheck);
+
+            services =
+              (lib.mapAttrs (
+                mainServiceName: serviceConfig:
+                let
+                  cfg = serviceConfig.healthcheck;
+                in
+                (mkIf (cfg.readiness-probe.enable) (
+                  let
+                    probeCfg = cfg.readiness-probe;
+                  in
+                  {
+                    serviceConfig.Type = lib.mkForce "notify";
+                    serviceConfig.TimeoutStartSec = lib.mkForce "infinity";
+
+                    # 2. Build the wrapper script that becomes the new ExecStart.
+                    serviceConfig.ExecStart =
+                      let
+                        scriptPath = makeBinPath (
+                          [
+                            pkgs.systemd
+                            pkgs.curl
+                            pkgs.gawk
+                          ]
+                          ++ (if cfg ? "path" then cfg.path else [ ])
+                          ++ (if serviceConfig ? "path" then serviceConfig.path else [ ])
```

Maybe prefer [`lib.optionals`](https://noogle.dev/f/lib/optionals)

```suggestion
                          ++ (lib.optionals (cfg ? "path") cfg.path)
                          ++ (lib.optionals (serviceConfig ? "path") serviceConfig.path)
```

or maybe this, as per <https://nix.dev/manual/nix/2.26/language/operators#attribute-selection>

```suggestion
                          ++ (cfg.path or [ ])
                          ++ (serviceConfig.path or [ ])
```